src/sys/timer.cxx

```diff
@@ -1,82 +1,72 @@
-#include <mpi.h>
 #include <bout/sys/timer.hxx>
+#include <mpi.h>
 
 using namespace std;
 
-Timer::Timer() {
-  timing = getInfo("");
-  timing->started = MPI_Wtime();
-  timing->running = true;
+Timer::Timer() : timing(getInfo("")) {
+  timing.started = MPI_Wtime();
+  timing.running = true;
 }
 
-Timer::Timer(const std::string &label) {
-  timing = getInfo(label);
-  timing->started = MPI_Wtime();
-  timing->running = true;
+Timer::Timer(const std::string& label) : timing(getInfo(label)) {
+  timing.started = MPI_Wtime();
+  timing.running = true;
 }
 
 Timer::~Timer() {
   double finished = MPI_Wtime();
-  timing->running = false;
-  timing->time += finished - timing->started;
+  timing.running = false;
+  timing.time += finished - timing.started;
 }
 
 double Timer::getTime() {
-  if(timing->running)
-    return timing->time + (MPI_Wtime() - timing->started);
-  return timing->time;
+  // If we've got a Timer by value, it must be running
```

Comment:
I'm not sure why you've got rid of the running check here but not in the next routine (`resetTime`). If `running = true` is the only possibility then why do we need it? I guess it's not the only possibility but in that case it's not obvious why we can avoid the check here (even with the comment).

Reply:
(also could/should this use `return getTime("")`?)

Reply:
Good point. I just spotted I could remove this one due to the coverage not hitting the other branch.

> (also could/should this use `return getTime("")` ?)

Not exactly, but a common function can be pulled out, yes